homeassistant/components/switch/arest.py

```diff
@@ -68,6 +70,69 @@ def is_on(self):
         """ True if device is on. """
         return self._state
 
+
+class ArestSwitchFunction(ArestSwitchBase):
+    """ Implements an aREST switch. Based on functions. """
+
+    def __init__(self, resource, location, name, func):
+        super().__init__(resource, location, name)
+        self._func = func
+
+        request = requests.get('{}/{}'.format(self._resource, self._func),
+                               timeout=10)
+
+        if request.status_code is not 200:
+            _LOGGER.error("Can't find function. Is device offline?")
+            return
+
+        try:
+            request.json()['return_value']
+        except KeyError:
+            _LOGGER.error("Function doesn't include return value. Is the function name correct.")
+        except ValueError:
+            _LOGGER.error("Function response invalid. Is the function name correct.")
+
+    def turn_on(self, **kwargs):
+        """ Turn the device on. """
+        request = requests.get('{}/{}'.format(self._resource, self._func),
+                               timeout=10, params={"params": "1"})
+
+        if request.status_code == 200:
+            self._state = True
+        else:
+            _LOGGER.error("Can't turn on function %s at %s. Is device offline?",
+                          self._func, self._resource)
+
+    def turn_off(self, **kwargs):
+        """ Turn the device off. """
+        request = requests.get('{}/{}'.format(self._resource, self._func),
+                               timeout=10, params={"params": "0"})
+
+        if request.status_code == 200:
+            self._state = False
+        else:
+            _LOGGER.error("Can't turn off function %s at %s. Is device offline?",
+                          self._func, self._resource)
+
+    def update(self):
```

Comment:
It looks like this one is similar to the pin based one. Can you set the url in the constructor and move this method to the base class?

If you define `turn_on_params` and `turn_off_params` in the constructor too, you'll be able to re-use `turn_on` and `turn_off` too.